src/thunderbots/software/ai/navigator/trespass.cpp

```diff
@@ -1,8 +1,59 @@
 #include "ai/navigator/trespass.h"
 
+#include <exception>
+
+#include "geom/circle.h"
+#include "geom/point.h"
+#include "util/math_functions.h"
+
 namespace Navigator
 {
     namespace Trespass
     {
-    }
+        // Don't know where to put this???
+        struct NanException : public std::exception
+        {
+            const char* what() const throw()
+            {
+                return "NaN operation exception";
+            }
+        };
+
+        unsigned int calcBinaryTrespassScore(Circle circle, Point point)
+        {
+            if (point.isnan() || circle.getOrigin().isnan())
+                throw NanException();
+
+            // The point is in the circle if the distance to
+            // the center of the circle is less than the radius
+            const double dx = point.x() - circle.getOrigin().x();
+            const double dy = point.y() - circle.getOrigin().y();
+            const double r  = circle.getRadius();
+
+            if (dx * dx + dy * dy > r * r)
+            {
+                // inside the circle
+                return 1;
+            }
+            else
+            {
+                return 0;
+            }
+        }
+
+        double calcLinearTrespassScore(Circle circle, Point point)
+        {
+            if (point.isnan() || circle.getOrigin().isnan())
+                throw NanException();
```

Comment:
Might be better to use `std::optional` here (return `std::nullopt` instead of throwing a custom exception).
@MathewMacDougall @garethellis0  for thoughts

Reply:
Yea I'd prefer an optional here since it's a little easier to handle than an exception.

Reply:
I've opened a discussion about this in slack (in the `#software` channel)

Reply:
Late to the party, but definitely agree that std::optional would be much simpler